src/bsp/tests/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+set(TEST_SOURCES
+    stm32f4/test.cpp
+)
+
+add_executable(test_stm32 ${TEST_SOURCES})
+
```

Comment:
Not gmock, but it's a template and we will eventually mock our calls, so I thinks it ok to let it. And yes we do need to include gtest since we are using the library to launch tests. Gtest-main allows to wrap all the tests in a main so you can easily create an executable.